guides/writing-addons/deprecations.md

```diff
@@ -0,0 +1,78 @@
+There will come a day when you need to make breaking changes to your addon. How can you guide your users through the changes? Follow these steps to create deprecation warnings, give your users enough information to resolve them, and test that your deprecations work as expected.
+
+## What is a deprecation warning?
+
+A deprecation warning is a message printed to the console that informs a developer that they need to refactor part of their app to accommodate upcoming breaking changes to a library they use.
+
+## When to write a deprecation warning
+
+Deprecation warnings should be written whenever you know that you want to remove or change a feature of your addon.
+
+You can create deprecations at any time, but to provide the best experience for other developers, it's a good idea to create a deprecation only once you have a workaround available.
+
+For example, a workaround could be instructions about how to use your other existing features to accomplish the same thing, or a new API method that developers should migrate to.
+
+## How to make a deprecation warning
+
+You can create a deprecation warning that looks and works the same as the deprecations used by Ember itself:
+
+```js
+import { deprecate } from '@ember/application/deprecations';
+// ...
+
+yourDeprecatedMethod() {
+    deprecate('You used yourDeprecatedMethod in my-addon-name, which is deprecated. Here are instructions to resolve this...', 
+        false,
+        {
```

Comment:
If possible, this POJO should have a `url` field that links to some "further reading" on your deprecation. The simple solution here is to link to your GH repo (e.g. a `docs/deprecations/some-id.md` file).

That URL is automatically printed to the console (no need to include it in the message), and tooling like the ember-inspector and ember-cli-deprecation-workflow collect the info and display it in a nice way.